resources/charts/bitcoincore/templates/configmap.yaml

```diff
@@ -7,6 +7,9 @@ metadata:
 data:
   bitcoin.conf: |
     {{- if eq .Values.chain "regtest" }}
+    regtest=1
+
+    {{ template "bitcoincore.check_semver.regtest" . }}
```

Comment:
we could define a _helper function or an entire _bitcoinconf.tpl just for managing all of the backwards compat, deprecated options, etc logic. so then you wouldn't have any of that logic in configmap.yaml, it would just call your helper function. this way we can reuse all of the config creation logic wherever we need it (altho it seems today we only need it in the config map).